src/libraries/Microsoft.PowerFx.Connectors/Shared/ServiceFunction.cs

```diff
@@ -389,12 +388,28 @@ public async Task<FormulaValue> InvokeAsync(FormulaValue[] args, CancellationTok
             {
                 throw new InvalidOperationException($"Function {Name} can't be invoked.");
             }
-            return await _invoker.InvokeAsync(args, cancel);
+
+            var result = await _invoker.InvokeAsync(args, cancel);
+            ExpressionError er = null;
+
+            if (result is ErrorValue ev && (er = ev.Errors.FirstOrDefault(e => e.Kind == ErrorKind.Network)) != null)
+            {
+                result = FormulaValue.NewError(
+                    new ExpressionError()
+                    {
+                        Kind = er.Kind,
+                        Severity = er.Severity,
+                        Message = $"{_invoker.Namespace.ToDottedSyntax()}.{_invoker.Name} failed: {er.Message}"
```

Comment:
>_invoker.Namespace. [](http://example.com/codeflow?start=37&length=19)

Can we avoid adding these to the IAsyncTexlFunction interface , and instead just pull them from this class's existing members?